DESIGN.md

```diff
@@ -266,6 +278,18 @@ This command checks to see if there are new versions of `swiftly` itself and upg
 
 `swiftly self-update`
 
+### Proxy command invocations to a requested toolchain
+
+Swiftly will create a set of symbolic links in its SWIFTLY_BIN_DIR during installation that point to the swiftly binary itself for each of the common toolchain commands, such as swift, swiftc, clang, etc. This mechanism will allows swiftly to proxy those command invocations to a requested toolchain. A toolchain can be requested in these ways in order of precedence:
+
+* Special toolchain selectors among the regular tool command-line arguments (e.g. `swift build +5.10.1`) with the special '+' prefix
+* The presence of a .swift-version file in the current working directory, or ancestor directory, with the required toolchain version
+* The swiftly default (in-use) toolchain set in the config.json by `swiftly install` or `swiftly use` commands
+
+In the first two cases, if there is no matching toolchain installed, swiftly will attempt to automatically install the requested toolchain and use it if the installation succeeeds.
```

Comment:
We'll need a way to disable this. People with a bad network connection will not be wanting to download 900MB of swift toolchain just to build a project.

Reply:
Yes, I'll update this since this has come up a few times. It could be a case of over automation. Instead, if there's no matching toolchain then the command fails. The `swiftly install` with no arguments can be used to install the currently selected toolchain with the full capabilities of that subcommand, including post install script, etc. Also, it's clear that an error is related to the install operation itself since that is the request.

Reply:
Could we also have a command line option to ignore the `.swift-version` file. This would make it easier for things like CI to test a library (which includes a `.swift-version` file) against multiple swift toolchains

Reply:
Yes, I think that this in this case the command-line selectors will take precedence over both the .swift-version file and the global default in-use toolchain from the config.json. The CI can do this to force specific toolchain(s):

```
swiftly install 5.10.1 # Install this specific toolchain
swift build +5.10.1   # Build with this toolchain
```